source/_posts/2020-09-16-supervisor-joins-the-party.markdown

```diff
@@ -0,0 +1,97 @@
+date: 2020-09-16 00:00:00
+date_formatted: "September 16, 2020"
+comments: true
+author: Joakim Sørensen
+author_twitter: ludeeus
+categories: Announcements
+---
+
+The Supervisor is responsible for managing your system so that you can manage your home. It’s the beating heart of the Home Assistant Operating System and it ensures that your system stays secure and up to date. When things do go wrong, the Supervisor is there to help you recover the system.
+
+Earlier this year, we limited the scope of what systems we support with the Supervisor. This has given us the opportunity to invest more time and resources into stability, usability improvements and new features.
+
+## Crash Reporting
+
+The Supervisor can now share anonymized diagnostics and crash information with the Home Assistant developers that work on the Supervisor. This option is, of course, opt-in and disabled by default. With this enabled, you submit issues without needing to open an issue!
+
+![diagnostics graphic](/images/blog/2020-09-16-supervisor-joins-the-party/diagnostics.png)
+
+Since we introduced this feature, over 25,000 events have been sent, generating around 120 unique issues. We have already fixed over 80 issues! With each release, fixes have been made as a result of these reports, making the Supervisor even more stable; This really shows how powerful it is.
+
+Big thanks to [Sentry][sentry], for powering and sponsoring this great feature.
+
+<blockquote>
+   "Accurate information is a key part of motivation." - Mary Ann Allison
+</blockquote>
+
+## Improve add-on availability with the new watchdog feature
+
+Add-ons for Home Assistant allow you to run third-party applications easily. These applications will benefit from the same management features also used to manage your Home Assistant installation.
+
+Add-on developers will be able to activate additional functionality for their add-on with the new “watchdog” add-on option. This enables application-level monitoring and allows the Supervisor to check if the add-on is behaving correctly using the ways specified by the developer.
+
+Not all add-ons have this feature and an add-on can still get themselves into problems that they can’t recover from, causing the application not to work as expected and crash. For this purpose, we have introduced a new watchdog toggle for advanced users. The functionality is similar to Docker’s health checks but operates outside of the container, making it a more robust option.
+
+![watchdog graphic](/images/blog/2020-09-16-supervisor-joins-the-party/watchdog.png)
+
+When the watchdog is enabled for an add-on, the Supervisor will automatically restart the add-on if it stops, regardless of the reason (crash/manual stop). The watchdog does not know if you’re testing an add-on or playing around with different options, and so it might restart when you don’t need it. You should only enable the watchdog after you are finished setting it up and want to make sure it’s running 24/7.
```

Comment:
```suggestion
When the watchdog is enabled for an add-on, the Supervisor will automatically restart the add-on if it stops, regardless of the reason (crash/manual stop). The watchdog does not know if you’re testing an add-on or playing around with different options, and so it might restart when you don’t need it. You should only enable the watchdog after you are finished setting the add-on up and want to make sure it’s running 24/7.
```